### PR TITLE
[#2297] Preview button now shows in header on project loaded from dashboard

### PR DESCRIPTION
**Pull request by mjschranz**
https://webmademovies.lighthouseapp.com/projects/65733/tickets/2297-opening-project-from-dashboard-doesnt-make-preview-button-visible

### CODE REVIEW

File: cornfield/app.js [hunk 1]
```diff
@@ -122,12 +122,34 @@ function writeEmbed( path, res, url, data, callback ) {
...
+      url;
+
+  if ( !id ) {
+    res.json( { error: "No Project ID specified" }, 404 );
+    return;
+  }
+
+  url = generatePublishUrl( id );
+
+  res.json( { error: "okay", publishURL: url }, 200 );
```
Comment (jbuck) [inline cornfield/app.js]: Change this to just be `url`, and you can remove the 200 status code, that's the default

File: cornfield/app.js [hunk 2]
```diff
@@ -122,12 +122,34 @@ function writeEmbed( path, res, url, data, callback ) {
...
+});
+
 app.post( '/api/publish/:id',
   filter.isLoggedIn, filter.isStorageAvailable, filter.isXHR,
   function publishRoute( req, res ) {
 
   var email = req.session.email,
-      id = req.params.id;
+      id = req.params.id,
+      publishURL = generatePublishUrl( id );
```
Comment (jbuck) [inline cornfield/app.js]: This isn't being used

File: cornfield/app.js [hunk 3]
```diff
@@ -122,12 +122,34 @@ function writeEmbed( path, res, url, data, callback ) {
...
+}
+
+app.get( '/api/publishurl/:id',
+  filter.isLoggedIn, filter.isStorageAvailable, filter.isXHR,
+  function( req, res ) {
+
+  var id = req.params.id,
+      url;
+
+  if ( !id ) {
```
Comment (jbuck) [inline cornfield/app.js]: in the regex you set, :id must always be set so you don't need to check for this condition

File: src/ui/header.js [hunk 4]
```diff
@@ -226,6 +226,13 @@ define([ "dialog/dialog", "util/lang", "ui/user-data", "ui/widget/tooltip" ],
     butter.listen( "ready", function() {
       if ( butter.project.name ) {
         _projectName.textContent = butter.project.name;
+
+        butter.cornfield.publishURL( butter.project.id, function( e ) {
+          if ( e.error === "okay" ) {
```
Comment (jbuck) [inline src/ui/header.js]: with the change above, `if ( url ) {`

File: cornfield/app.js [hunk 5]
```diff
@@ -122,12 +122,34 @@ function writeEmbed( path, res, url, data, callback ) {
   });
 }
 
+function generatePublishUrl( id ) {
+  return PUBLISH_PREFIX_V + "/" + id + ".html";
+}
+
+app.get( '/api/publishurl/:id',
```
Comment (jbuck) [inline cornfield/app.js]: Move this into `routes/index.js`

File: cornfield/app.js (hunk 2 above)
Comment (mjschranz) [inline cornfield/app.js]: Yes it was. Look further down in teh code.

Comment (mjschranz) [inline cornfield/app.js]: Rather, it was supposed to be.

File: cornfield/app.js [hunk 6]
```diff
@@ -233,14 +241,10 @@ app.post( '/api/publish/:id',
       }
       popcornString += '</script>\n';
 
-      data = startString + baseString + templateScripts + externalAssetsString +
-             data.substring( headEndTagIndex, bodyEndTagIndex ) +
-             popcornString + data.substring( bodyEndTagIndex );
-
```
Comment (mjschranz) [inline cornfield/app.js:239]: I removed this because our templating jade files cause this to actually not be used at all anymore.

**Comment by jbuck** [comment]
Closing in favour of #1126.

(holy crap we've had 100 pull requests in 18 days)

**Pull request closed without merging**